tests-e2e/cypress/support/plugin/ui_commands.js

```diff
@@ -308,11 +308,23 @@ Cypress.Commands.add('interceptTelemetry', () => {
     cy.intercept('/plugins/playbooks/api/v0/telemetry').as('telemetry');
 });
 
-// expectTelemetryToBe expects to find the given telemetry events in the order given among the
+const defaultExpectTelemetryToContainOptions = {
+    waitForCalls: 'auto',
+};
+
+// cy.expectTelemetryToContain expects to find the given telemetry events in the order given among the
 // recorded telemetry. It doesn't fail if other telemetry events happen to occur in between.
-Cypress.Commands.add('expectTelemetryToBe', (items) => {
-    // Wait for at least as many telemetry events as requested.
-    items.forEach(() => cy.wait('@telemetry'));
+Cypress.Commands.add('expectTelemetryToContain', (items, opts) => {
+    const options = {...defaultExpectTelemetryToContainOptions, ...opts};
+
+    // Wait for at least as many telemetry events as requested if auto, or explicit number is passed.
```

Comment:
```suggestion
// Wait for at least as many telemetry events as requested if auto, or explicit number if passed.
```

Would a test that passes a number here likely be brittle in the face of unexpected code changes?

Reply:
The passed number is the number of guaranteed requests to be waited for.

If I want to test follow&unfollow events but the pageview is additionally tracked, I should wait for 3 events to ensure my desired events are captured. If I stopped tracking the pageview the test will fail because we can not satisfy the third wait. 

I could add a comment to the passed param about which events are waited but not asserted, in order to solve a potential error in an easier way. But that is only a comment...


What I would eventually want, but I still didn't completely solve:
- [X] validate just the properties passed 
- [X] check that some events triggered
- [ ] wait automatically for all events
- [ ] ignore the order of the events

Reply:
As we add more and more events, it's not very nice to assert all events that trigger every time we want to ensure that one event is triggered.